### DISCUSSION TITLE
add child as first

### DISCUSSION THREAD

**Discussion by Olifair** [freeplane/freeplane#1707] [Help]
Hello, 
I just wondered (and looked through the preferences) if it is possible that child nodes are added as first (on top) and not as last (at bottom) ?   for Todo lists I would find it more intuitive that newest (often most urgent) are added on top.  I haven't found a preference related to this, so perhaps this is not an option. 
Thanks in advance
Oliver

**Comment by euu2021** [comment]
I don't remember seeing this option. It's interesting to note that, in Outline view (View->View settings->Outline view) the child nodes are added to the top.

**Comment by Olifair** [comment]
Yes, you are right, in the outline view, a newly inserted child node is at the top. However when you drag and drop a node on a parent node on its right side (distal side of the edge), then it is still added at the end.

**Reply by abc16361** [reply]
You can create a script to insert a child node and use` moveTo(node, 0) `within the script to place the child on top.

```
// @ExecutionModes({ON_SINGLE_NODE})

def newChild = node.createChild()
newChild.moveTo(node, 0)
```

**Comment by Olifair** [comment]
thanks @abc16361 I appreciate the script (and will tick it away for later use) . But I can actually already do this in the outline look, new child-nodes created there are on top. What I'm really looking for is that this is also working with drag and drop. Let me explain : I have a map for organising my things to do (a kind of First Things First approach). So I have nodes for things that need to be done today, this week, next week, within this month, later. I review these lists regularly and drag and drop my todo-nodes to a new category. Dropping a Todo from next week to this week means that I will have to do this rapidly, and this I would like to have this node on top (and not at the bottom)....

**Comment by i-plasm** [comment]
The preference already exists in: `Preferences…->Behaviour->Place new branches` . Just select option called `First`

**Comment by Olifair** [comment]
@i-plasm  Thanks, how could I have overlooked this, it is actually the first option . However it also only works for newly created child nodes, not when dragging notes, these will still be added at the end. But thanks anyway.

**Reply by dpolivaev** [reply]
It is because when you drag and drop you can easily drop at the top of the first child

**Reply by i-plasm** [reply]
For that situation (dragging), besides Dimitry's suggestion above, if your dragged nodes appear at the very end, 
 you can press `CTRL + DOWN ARROW` (assuming they are selected) and they will be sent to top.

**Reply by euu2021** [reply]
Related discussion: #1357

**Comment by Olifair** [comment]
thanks @dpolivaev  , yes, that would be a solution.  and also thanks @i-plasm , that's also a very usefull idea, I hadn't thought that moving nodes with CTRL+ arrows would send the last to the top. 
I don't remember for how many years I have been using FreePlane now (my oldest maps seem to be from 2009), but I'm still learning about options and ways to do things. It's just incredible how versatile FreePlane is.
